src/matrixClass.ts

```diff
@@ -0,0 +1,317 @@
+import {Position} from './Position'
+
+export class Matrix{
+    public board:number[][] = [];
+    /*
+    0 = uncharted
+    1 = hit
+    2 = miss
+    3 = suggested target
+    4 = blocked out no ship area
+    */
+    public ships:boolean[] = [true,true,true,true,true];
+    /*
+    index = length
+    0 = 2
+    1 = 3
+    2 = 3
+    3 = 4
+    4 = 5
+     */
+
+
+    
+    public returnUnsunkShot(){
+        for(let row =0; row < this.board.length; row ++){
+            for(let col = 0; col < this.board.length; col ++){
+                if(this.board[row][col] == 3)return this.positionFromCoords(row, col);
+            }
+        }
+        // should never reach hear;
+        console.log("never logged");
```

Comment:
This is by definition an error state, so why not actually create an error, something like, `throw new Error("Couldn't get unsunk shot"`. That way, iut's more obvious in the console, & you can remove your comment/log!